_posts/2018-08-27-deploy-digitalocean-ansible.md

```diff
@@ -0,0 +1,195 @@
+
+```bash
+$ export DOTOKEN=<API token> && \
+  curl -X GET -H 'Content-Type: application/json' \
+   -H 'Authorization: Bearer '$DOTOKEN \
+   "https://api.digitalocean.com/v2/account/keys" 2>/dev/null \
+   |  jq  '.ssh_keys[] | {name: .name, id: .id}'
+
+{
+"name": "openfaas_key",
+"id": 12345000
+}
+```
+
+In this example there is only one key, and so the `sshid` of interest is `12345000`.  Where there are multiple keys configured obtain the id associated with the name entered when uploading the key.
+
+## Configure the Playbook
+
+Clone the playbook:
+
+```bash
+$ mkdir -p ~/openfaas && \
+  cd ~/openfaas && \
+  git clone git@github.com:rgee0/openfaas-on-digitalocean.git .
+```
+
+The playbook definition can be found in `site.yml`. The API key and chosen sshid now need adding to the playbook as variables in the `create_droplet` role used in the first play.
+
+Navigate to `create_droplet/vars/main.yml` and add the two earlier obtained values as the `token` and `sshid` variables.  Although the playbook has been tested with the default values it is also possible to adjust the [size of the machine](https://developers.digitalocean.com/documentation/changelog/api-v2/new-size-slugs-for-droplet-plan-changes/) to be deployed, the region and the OS image.
+
+The resulting `main.yml` should look similar to this:
+
+```yaml
+---
+  token: BE39Z34848C963914765D5DAB11DB123CB5D28F96067E538ZA
+  sshid: 12345000
+  size_slug: s-2vcpu-4gb
+  region: lon1
+  image: ubuntu-18-04-x64
+```
+
+### Run the Playbook
+Having added the required detail to `create_droplet/vars/main.yml` then the playbook is ready to be run.  Running the playbook will call upon a Docker image, `rgee0/ansible-playbook:2.6.0`, which offers a ready-made Ansible environment.  Three volumes are mounted, two make the SSH key-pair available to Ansible - the values represented below are consistent with locations used earlier, ensure these are changed if different locations / names were used.  The final volume mounts the project directory into the container so that Ansible can access the playbook.
+
+> To execute from a Raspberry Pi 3 then use `rgee0/ansible-playbook:2.6.0-arm32v7` instead
+
+All that remains is to choose one of the two orchestrators and pass the value as a extra variable:
+
+Kubernetes
+
+```bash
+$ cd ~/openfaas && \
+  docker run --rm -it \
+  -v $(pwd):/ansible/playbooks \
+  -v ~/.ssh/id_rsa_blog:/root/.ssh/id_rsa \
+  -v ~/.ssh/id_rsa_blog.pub:/root/.ssh/id_rsa.pub \
+  rgee0/ansible-playbook:2.6.0 site.yml -e "orchestrator=k8s"
+```
+
+Swarm 
+
+```bash
+$ cd ~/openfaas && \
+  docker run --rm -it \
+  -v $(pwd):/ansible/playbooks \
+  -v ~/.ssh/id_rsa_blog:/root/.ssh/id_rsa \
+  -v ~/.ssh/id_rsa_blog.pub:/root/.ssh/id_rsa.pub \
+  rgee0/ansible-playbook:2.6.0 site.yml -e "orchestrator=swarm"
+```
+
```

Comment:
Call out what you're changing

> orchestrator=swarm

Mention that you're mounting the public and private SSH key here. Also link to how the container was built (should anyone trust you with their private SSH key especially if the image was built manually?) Can you change to a Travis build in time to publish?